test/units/utils/collection_loader/test_collection_loader.py

```diff
@@ -37,7 +38,11 @@ def test_find_module_py3():
     # setuptools may fall back to find_module on Python 3 if find_spec returns None
     # see https://github.com/pypa/setuptools/pull/2918
     assert path_hook_finder.find_spec('missing') is None
-    assert path_hook_finder.find_module('missing') is None
+    if sys.version_info < (3, 12):
+        # removed in Python 3.12
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", DeprecationWarning)
```

Comment:
@s-hertel could you replace this with a decorator following pytest's instructions and make it more accurate?

It'd looks like this:
```python
@pytest.mark.filterwarnings(
    'ignore:'
    r'the load_module\(\) method is deprecated and slated for removal '
    r'in Python 3\.12; use exec_module\(\) instead'
    ':DeprecationWarning',
)
```

Ref: https://docs.pytest.org/en/7.1.x/how-to/capture-warnings.html#pytest-mark-filterwarnings

Reply:
If we're filtering warnings for the entire function using a decorator, we should probably split out the assert into its own test so the scope of the warning suppression is limited to the assert it's intended for.

Reply:
good idea, split into two tests